### PR TITLE
Hide action button when no action provided in notification banner IOS-168

### PR DESCRIPTION
**Pull request by pronebird**
1. Update constraints that pin the body label to the trailing layout margin when no action provided, otherwise pin the trailing edge to the leading edge of action button.
2. Add helper to store constraint into array `NSLayoutConstraint.store(into: inout [NSLayoutConstraint])` that provides chained call semantics.
3. Modernize constraint creation by using `addConstrainedSubviews` and autolayout helpers we use to reduce amount of boilerplate code.
4. Add missing trailing anchor for title label.
    HasAction
    NoAction
5. Set action button opacity to 80%
6. Add system spacing button the body label and action button.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)

### CODE REVIEW

**Review by rablador — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: all files reviewed, 1 unresolved discussion (waiting on @pronebird)

___
*[`ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift` line 93 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695#-NVZnrcm7EdgOOlWElZb:-NVZnrcm7EdgOOlWElZc:bqzi6qp) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3362bb32d2a4386412dc2fe82cddbe7677e003d1/ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift#L93)):*
> ```Swift
>     }
> 
>     private var actionButtonVisibleConstraints: [NSLayoutConstraint] = []
> ```

Can't we put `bodyLabel` and `actionButton` inside a stackview instead? That way we don't need to handle constraints whether `actionButton` is visible or not.

**Review by pronebird — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: all files reviewed, 1 unresolved discussion (waiting on @rablador)

___
*[`ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift` line 93 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695#-NVZnrcm7EdgOOlWElZb:-NVZqSZWEVpboMl996lk:b-r8w1o4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3362bb32d2a4386412dc2fe82cddbe7677e003d1/ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift#L93)):*
<details><summary><i>Previously, rablador (Jon Petersson) wrote…</i></summary><blockquote>

Can't we put `bodyLabel` and `actionButton` inside a stackview instead? That way we don't need to handle constraints whether `actionButton` is visible or not.
</blockquote></details>

I have tried and the whole thing fell apart. Let me try again.

**Review by pronebird — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @rablador)

___
*[`ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift` line 93 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695#-NVZnrcm7EdgOOlWElZb:-NVZxAVr3TsOvLe4YTyJ:b-9kp2d4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3362bb32d2a4386412dc2fe82cddbe7677e003d1/ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift#L93)):*
<details><summary><i>Previously, pronebird (Andrej Mihajlov) wrote…</i></summary><blockquote>

I have tried and the whole thing fell apart. Let me try again.
</blockquote></details>

OK done seems to work now.

**Review by rablador — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by mojganii — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: all files reviewed, 2 unresolved discussions (waiting on @pronebird)

___
*[`ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift` line 98 at r4](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695#-NW7dKwc8tt2V21AOGUi:-NW7dKwc8tt2V21AOGUj:b-kr8jv3) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/754d2f53f4488e439d0db04442cab9bef045c0fd/ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift#L98)):*
> ```Swift
>         super.init(frame: frame)
> 
>         actionButton.addTarget(self, action: #selector(didPress), for: .touchUpInside)
> ```

maybe it's better to break this part of code into two functions.
1- setupViews
2- setupConstraints

**Review by rablador — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: all files reviewed, 2 unresolved discussions (waiting on @mojganii)

**Review by pronebird — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @mojganii and @rablador)

___
*[`ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift` line 98 at r4](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695#-NW7dKwc8tt2V21AOGUi:-NWINZ-j9m9rxWzcKPCX:b-gkscim) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/754d2f53f4488e439d0db04442cab9bef045c0fd/ios/MullvadVPN/Notifications/UI/NotificationBannerView.swift#L98)):*
<details><summary><i>Previously, pronebird (Andrej Mihajlov) wrote…</i></summary><blockquote>

Where would you put the following lines that we have? Is it 1. or 2.?

```swift
addConstrainedSubviews([backgroundView]) {
     backgroundView.pinEdgesToSuperview()
}
```
</blockquote></details>

Broke down `init()` on a few functions. Hope that looks better for your taste.

**Review by rablador — APPROVED** [review]
> Reviewed 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4695)* status: all files reviewed, 2 unresolved discussions (waiting on @mojganii)
